tutorials/mission_specific_analyses/nicer/data-analysis-nicer.md

```diff
@@ -0,0 +1,598 @@
+
+```{code-cell} python
+query = (
+    "SELECT * "
+    "from {c} as cat "
+    "where cat.obsid='{oi}'".format(oi=OBS_ID, c=HEASARC_TABLE_NAME)
+)
+
+query
+```
+
+The query is then executed, and the returned value is converted to an AstroPy table (necessary for the next step):
+
+```{code-cell} python
+obs_line = Heasarc.query_tap(query).to_table()
+obs_line
+```
+
+Identifying the 'data link' that we need to download the data files is now as simple as passing the query return
+to the `locate_data` method of `Heasarc`:
+
+```{code-cell} python
+data_links = Heasarc.locate_data(obs_line, HEASARC_TABLE_NAME)
+data_links
+```
+
+This data link can be passed straight into the `download_data` method of `Heasarc`, and our observation data files
+will be downloaded into the directory specified by `ROOT_DATA_DIR`.
+
+```{code-cell} python
+# Heasarc.download_data(data_links, host="sciserver", location=ROOT_DATA_DIR)
+Heasarc.download_data(data_links, host="aws", location=ROOT_DATA_DIR)
+
+# We remove the existing cleaned event list directory from the data we just downloaded
+shutil.rmtree(os.path.join(ROOT_DATA_DIR, OBS_ID, "xti", "event_cl"))
+```
+
+## 2. Processing and cleaning NICER observations
+
+```{danger}
+NICER level-2 processing now **requires** up-to-date geomagnetic data
+([see this for a discussion](https://heasarc.gsfc.nasa.gov/docs/nicer/analysis_threads/geomag/)); we used a HEASoftPy
+tool (nigeodown) to download the latest geomagnetic data in the 'Global setup: Configuration' section near the top of
+this notebook. You should make sure to regularly update your geomagnetic data!
+```
+
+Next up, we're going to run the `nicerl2` pipeline to process and clean the raw NICER data; this will render it
+ready for scientific use. NICER pipelines are implemented in HEASoft, and as such we can make use of interfaces
+built into HEASoftPy to run them in this notebook, rather than in the command line.
+
+This pipeline is designed to produce "level 2" data products and includes steps for standard calibration,
+screening, and filtering of data. The outputs are an updated filter file and a cleaned event list.
+
+***It is worth applying `nicerl2` to any observation that you know was last processed with an older calibration!***
+
+```{note}
+The `filtcolumns` argument to `nicerl2` will default to the latest version of the standard columns (V6 as of the
+20th October 2025). You may also set it manually (as we do here) for backwards compatibility reasons - see the
+'Selecting filter file columns' section of [the nicerl2 help file](https://heasarc.gsfc.nasa.gov/docs/software/lheasoft/help/nicerl2.html) for more information.
+```
+
+```{code-cell} python
+OBS_ID_PATH = os.path.join(ROOT_DATA_DIR, OBS_ID)
+
+# Run the task
+out = nicerl2(
+    indir=OBS_ID_PATH,
+    geomag_path=GEOMAG_PATH,
+    filtcolumns="NICERV5",
+    clobber=True,
+    noprompt=True,
+    allow_failure=False,
+)
+```
+
+```{error}
```

Comment:
I suggest we just set `evtsuffix` ourselves in nicerl2 and other tasks and avoid this part. We can say (in a tip or a warning) we are doing this because there is a bug in heasoft 6.36, and that it is generally not needed.